core/market/decentralized/src/market.rs

```diff
@@ -102,8 +102,8 @@ impl MarketService {
             .extend(rest_api::requestor::register_endpoints)
     }
 
-    pub async fn get_offers(&self) -> Result<Vec<Offer>, MarketError> {
-        Ok(self.matcher.store.get_offers().await?)
+    pub async fn get_offers(&self, id: Option<Identity>) -> Result<Vec<Offer>, MarketError> {
```

Comment:
we do not need Option at this level. Pass`Identity` from REST API and wrap it here into `Some`